src/components/TokenBottomSheet.tsx

```diff
@@ -218,68 +229,67 @@ function TokenBottomSheet<T extends TokenBalance>({
-              onSelectChip={handleToggleFilterChip}
-              primaryColor={colors.successDark}
-              secondaryColor={colors.successLight}
-              style={styles.filterChipsCarouselContainer}
-              forwardedRef={filterChipsCarouselRef}
-            />
-          )}
-        </>
-      }
       onOpen={handleOpen}
       onClose={handleClose}
-      testId="TokenBottomSheet"
     >
-      {tokenList.length == 0 ? (
-        searchEnabled || filterChips.length > 0 ? (
-          <NoResults searchTerm={searchTerm} activeFilters={activeFilters} />
-        ) : null
-      ) : (
-        tokenList.map((tokenInfo, index) => {
-          return (
-            // Duplicate keys could happen with token.address
-            <React.Fragment key={`token-${tokenInfo.tokenId ?? index}`}>
-              <TokenOptionComponent
-                tokenInfo={tokenInfo}
-                onPress={onTokenPressed(tokenInfo)}
-                index={index}
-                showPriceUsdUnavailableWarning={showPriceUsdUnavailableWarning}
-              />
-            </React.Fragment>
-          )
-        })
-      )}
-    </BottomSheet>
+      <View style={styles.headerContainer}>
+        <Text style={[styles.title, titleStyle]}>{title}</Text>
+        {searchEnabled && (
+          <SearchInput
+            placeholder={t('tokenBottomSheet.searchAssets') ?? undefined}
+            value={searchTerm}
+            onChangeText={(text) => {
+              setSearchTerm(text)
+              sendAnalytics(text)
+            }}
+            style={styles.searchInput}
+            returnKeyType={'search'}
+            // disable autoCorrect and spellCheck since the search terms here
+            // are token names which autoCorrect would get in the way of. This
+            // combination also hides the keyboard suggestions bar from the top
+            // of the iOS keyboard, preserving screen real estate.
+            autoCorrect={false}
+            spellCheck={false}
+          />
+        )}
+        {filterChips.length > 0 && (
+          <FilterChipsCarousel
+            chips={filters}
+            onSelectChip={handleToggleFilterChip}
+            primaryColor={colors.successDark}
+            secondaryColor={colors.successLight}
+            style={styles.filterChipsCarouselContainer}
+            forwardedRef={filterChipsCarouselRef}
+          />
+        )}
+      </View>
+      <BottomSheetFlatList
+        ref={tokenListRef}
+        testID="TokenBottomSheet"
+        data={tokenList}
+        keyExtractor={(item) => item.tokenId}
+        contentContainerStyle={styles.tokenListContainer}
+        renderItem={({ item, index }) => (
+          <TokenOptionComponent
+            tokenInfo={item}
+            onPress={onTokenPressed(item)}
+            index={index}
+            showPriceUsdUnavailableWarning={showPriceUsdUnavailableWarning}
+          />
+        )}
+        initialNumToRender={8}
+        maxToRenderPerBatch={8}
```

Comment:
Looks like this should be based on the screen size and item height to avoid having some initial blanks, on bigger screens.

Reply:
Sounds good, I didn't mean we should really measure, but we could have used a hardcoded item height as a quick shortcut too.